src/monitoring_tests/buffers_monitoring_test.py

```diff
@@ -0,0 +1,100 @@
+from src.constants import (
+    BUFFER_INTERVAL,
+    header,
+    REQUEST_TIMEOUT,
+    BUFFER_VALUE_THRESHOLD,
+)
+
+
+class BuffersMonitoringTest(BaseTest):
+    """
+    This test checks the value of the settlement contract buffers
+    every 150 settlements and generates an alert if it is higher than 200_000 USD.
+    Price feeds from ethplorer and coingecko (as backup) are used.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.counter: int = 0
+        self.buffers_value: float = 0.0
+
+    def compute_buffers_value(self) -> bool:
+        """
+        Evaluates current state of buffers.
+        """
+        # get all token balances of the smart contract
+        try:
+            resp = requests.get(
+                "https://api.ethplorer.io/\
+                    getAddressInfo/\
+                    0x9008D19f58AAbD9eD0D60971565AA8510560ab41?\
+                    apiKey=freekey",
+                headers=header,
+                timeout=REQUEST_TIMEOUT,
+            )
+            rsp = resp.json()
+
+            value_in_usd = 0.0
+            for token in rsp["tokens"]:
+                balance = token["balance"]
+                decimals = int(token["tokenInfo"]["decimals"])
+                if token["tokenInfo"]["price"] is not False:
+                    price_in_usd = token["tokenInfo"]["price"]["rate"]
+                    token_buffer_value_in_usd = (
+                        balance / 10**decimals
+                    ) * price_in_usd
+                    # in case some price is way off and it blows a lot the total value held in the
+                    # smart contract we use a second price feed, from coingecko, to correct in case
+                    # the initial price is indeed off
+                    if token_buffer_value_in_usd > 10000:
+                        coingecko_resp = requests.get(
+                            "https://api.coingecko.com/\
+                                api/v3/simple/token_price/\
+                                ethereum?contract_addresses="
+                            + token["tokenInfo"]["address"]
+                            + "&vs_currencies=usd",
+                            headers=header,
+                            timeout=REQUEST_TIMEOUT,
+                        )
+                        coingecko_rsp = coingecko_resp.json()
+                        coingecko_price_in_usd = coingecko_rsp[
+                            token["tokenInfo"]["address"]
+                        ]["usd"]
+                        coingecko_value_in_usd = (
+                            balance / 10**decimals
+                        ) * coingecko_price_in_usd
+                        if coingecko_value_in_usd < token_buffer_value_in_usd:
+                            token_buffer_value_in_usd = coingecko_value_in_usd
+                    value_in_usd += token_buffer_value_in_usd
+            self.buffers_value = value_in_usd
```

Comment:
`self.buffers_value` does not seem to be used as class variable. One could also just use `value_in_usd` in the conditional below.

Reply:
Fair point. I introduced it mostly to have it there just in case, but you are right that it is not needed. I am removing this for now and might reintroduce it later in case there is an actual use case for that.